src/mcts/storage.cc

```diff
@@ -3,44 +3,51 @@
 
 namespace mcts {
 
-static std::mutex freeStoragesMutex;
-static std::list<Storage*> freeStorages;
+static std::mutex storagesMutex;
+static std::vector<std::unique_ptr<Storage>> allStorages;
+static std::vector<Storage*> freeStorages;
+
 
 Node* Storage::newNode() {
-  if (chunkIndex >= chunks.size()) {
-    Node* newChunk = (Node*)std::aligned_alloc(128, sizeof(Node) * chunkSize);
-    new (newChunk) Node[chunkSize];
-    for (size_t i = 0; i != chunkSize; ++i) {
-      newChunk[i].setStorageAndId(this, i);
+  if (chunks_it == chunks.end()) {
```

Comment:
The logic is basically the same, but now it operates on a vector of `unique_ptr` instead of naked pointers.

Reply:
ok